app/assets/javascripts/discourse/app/components/create-account.js

```diff
@@ -4,13 +4,39 @@ import cookie from "discourse/lib/cookie";
 export default Component.extend({
   classNames: ["create-account"],
 
+  userInputFocus(e) {
+    let label = e.target.parentElement.previousElementSibling;
+    if (!label.classList.contains("value-entered")) {
+      label.classList.toggle("value-entered");
+    }
+  },
+
+  userInputFocusOut(e) {
+    let label = e.target.parentElement.previousElementSibling;
+    if (
+      e.target.value.length === 0 &&
+      label.classList.contains("value-entered")
+    ) {
+      label.classList.toggle("value-entered");
+    }
+  },
+
   didInsertElement() {
     this._super(...arguments);
 
     if (cookie("email")) {
       this.set("email", cookie("email"));
     }
 
+    let userTextFields = document
+      .getElementsByClassName("user-fields")[0]
+      .getElementsByClassName("ember-text-field");
+
+    for (let element of userTextFields) {
```

Comment:
why not forEach ?

Reply:
I do not believe `forEach` exists on the element node list.

Reply:
It does exist for `querySelectorAll` but not for `getElementsByClassName` which is the way I am finding the specific text fields within the user field area.